projects/packages/jetpack-mu-wpcom/src/features/wpcom-blocks/code/block-definition/block-definition.tsx

```diff
@@ -0,0 +1,517 @@
+	 * defined elsewhere.
+	 *
+	 * For example:
+	 * `{ colorComment?: string; }`
+	 * becomes
+	 * `{ '--colorComment': string | undefined; }`
+	 */
+	[ key in `--${ keyof Pick<
+		Attributes,
+		Extract< keyof Attributes, `color${ Capitalize< string > }` >
+	> }` ]-?: string | undefined;
+} & {
+	'--line-numbers-start-at'?: string;
+	'--line-number-gutter-width'?: string;
+	'--colorBackground'?: string;
+	'--colorText'?: string;
+};
+
+/**
+ * Transforms attributes into CSS custom properties for inline style use.
+ *
+ * @param attributes - Block attributes.
+ * @return CSS style object.
+ */
+function blockStyle( attributes: Attributes ): BlockStyleProperties {
+	const properties: BlockStyleProperties = {
+		'--colorComment': attributes.colorComment,
+		'--colorKeyword': attributes.colorKeyword,
+		'--colorBoolean': attributes.colorBoolean,
+		'--colorLiteral': attributes.colorLiteral,
+		'--colorString': attributes.colorString,
+		'--colorSpecialString': attributes.colorSpecialString,
+		'--colorMacroName': attributes.colorMacroName,
+		'--colorVariableDefinition': attributes.colorVariableDefinition,
+		'--colorTypeName': attributes.colorTypeName,
+		'--colorClassName': attributes.colorClassName,
+		'--colorInvalid': attributes.colorInvalid,
+	} satisfies React.CSSProperties;
+
+	if ( attributes.showLineNumbers && attributes.tokenizedLines.length ) {
+		const maxLineNumberWidth =
+			Math.floor(
+				Math.log10( attributes.lineNumbersStartAt + ( attributes.tokenizedLines.length - 1 ) )
+			) + 1;
+		properties[ '--line-numbers-start-at' ] = String( attributes.lineNumbersStartAt );
+		properties[ '--line-number-gutter-width' ] = `${ maxLineNumberWidth }ch`;
+	}
+
+	if ( attributes.backgroundColor ) {
+		properties[
+			'--colorBackground'
+		] = `var( --wp--preset--color--${ attributes.backgroundColor } )`;
+	} else if ( attributes.style?.color?.background ) {
+		properties[ '--colorBackground' ] = attributes.style.color.background;
+	}
+	if ( attributes.textColor ) {
+		properties[ '--colorText' ] = `var( --wp--preset--color--${ attributes.textColor } )`;
+	} else if ( attributes.style?.color?.text ) {
+		properties[ '--colorText' ] = attributes.style.color.text;
+	}
+
+	return properties;
+}
+
+/**
+ * Perform HTML encoding.
+ *
+ * This is designed to encode text for HTML printing. The characters `&`, `<`, `>`,
+ * and `[` are HTML encoded.
+ *
+ * `[` is HTML encoded to prevent shortcodes from being processed.
+ *
+ * A numeric encoding is used for `&` as a workaround for
+ * {@link https://core.trac.wordpress.org/ticket/63630|Trac 63630}.
+ * The issue should be fixed in WordPress 6.9.
+ * @param content - Original content.
+ * @return Encoded content.
+ */
+function htmlEncode( content: string ): string {
+	return content
+		.replaceAll( '&', '&#38;' )
+		.replaceAll( '<', '&lt;' )
+		.replaceAll( '>', '&gt;' )
+		.replaceAll( '[', '&#91;' )
+		.replaceAll( "'", '&#39;' )
+		.replaceAll( '"', '&#22;' );
```

Comment:
`22` is in hex, this should be the decimal `34`:
```suggestion
		.replaceAll( '"', '&#34;' );
```